### PR TITLE
Feature/dsg 902/data validation

### PR DESCRIPTION
**Pull request by jcsuarezm**
## Types of changes

Patch
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation change (documentation only, will not require redeployment)
- [ ] Refactor or cleanup (functionality unchanged)
Minor
- [x] New feature (non-breaking change which adds functionality)
Major
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

- [x] I have confirmed all acceptance criteria defined by the story have been met

### CODE REVIEW

File: service/src/main/java/io/nuvalence/workmanager/service/mapper/EntityMapper.java [hunk 1]
```diff
@@ -123,53 +123,76 @@ public Map<String, Object> convertAttributesToGenericMap(final DynamicEntity ent
...
-                            attributeName,
-                            convertSingleValueToEntity(
-                                    schema, dynaProperty.getType(), attributeName, value));
+                    list = convertListValueToEntity(schema, key, (List<Object>) data.get(key));
+                }
+                entity.set(key, list);
+            } else {
+                if (!dynaPropertyType.isAssignableFrom(element.getClass())) {
+                    try {
+                        element = DataConversionSupport.convert(element, dynaPropertyType);
```
Comment (jgnuvalence) [inline service/src/main/java/io/nuvalence/workmanager/service/mapper/EntityMapper.java:180]: this works for me for all cases i tried except for LocalDate:

schema:
```
{
      "name": "dateOfBirth",
      "type": "LocalDate",
      "attributeConfigurations": []
    }

```
Calling the endpoint
image

Result of the evaluation in this line is null, not an exception:
image


The result is 200, but it should have been 400 and return a message saying that the format of dateOfBirth must be a date:
image

**Review by jgnuvalence — CHANGES_REQUESTED** [review]
I left some comments

File: service/src/main/java/io/nuvalence/workmanager/service/service/TransactionService.java [hunk 2]
```diff
@@ -746,4 +746,30 @@ public AuditEventId postAuditEventForTransactionCreated(Transaction transaction)
...
+                if (entry.getValue() instanceof Map
+                        && existing.get(entry.getKey()) instanceof Map) {
+                    unifyAttributeMaps(
+                            (Map<String, Object>) entry.getValue(),
+                            (Map<String, Object>) existing.get(entry.getKey()));
+                } else {
+                    existing.put(entry.getKey(), entry.getValue());
+                }
+            } else {
+                existing.put(entry.getKey(), entry.getValue());
```
[the remark this answers is not in the document]
Comment (jcsuarezm) [reply service/src/main/java/io/nuvalence/workmanager/service/service/TransactionService.java:767]: Thanks @jgnuvalence! The conditions have been unified.

File: service/src/main/java/io/nuvalence/workmanager/service/mapper/EntityMapper.java [hunk 3]
```diff
@@ -123,53 +123,76 @@ public Map<String, Object> convertAttributesToGenericMap(final DynamicEntity ent
...
+                        String.format("Key not found in schema: '%s'.", key));
+            }
+
+            Object element = data.get(key);
+            Class<?> dynaPropertyType = currentDynaProperty.getType();
+
+            if (DynamicEntity.class.isAssignableFrom(dynaPropertyType)) {
+                if (!Map.class.isAssignableFrom(element.getClass())) {
+                    throw new BusinessLogicException(
+                            String.format("Invalid type for key: '%s'. It should be a Map.", key));
```
[the remark this answers is not in the document]
Comment (jcsuarezm) [reply service/src/main/java/io/nuvalence/workmanager/service/mapper/EntityMapper.java]: The message has been changed, now it uses composite object instead, as follows: 
image

File: service/src/main/java/io/nuvalence/workmanager/service/mapper/EntityMapper.java (hunk 1 above)
Comment (jcsuarezm) [reply service/src/main/java/io/nuvalence/workmanager/service/mapper/EntityMapper.java]: Thank you! This situation has been corrected by adding a null validation, this was tested as follows: 
image
Document validation was also tested, as follows: 
image

**Review by jgnuvalence — APPROVED** [review]

**Pull request merged**